### PR TITLE
Feature/477/demo ar workflows gitlab

### LINKED ISSUE #477: [FEATURE] Auto Remediation demo based on Workflows
**Is your feature request related to a problem? Please describe.**
Currently, only an auto remediation demo with problem notifications and AWX is available. With the availability of gen 3, the auto rem demo shall also be updated to use new gen 3 features, i.e. Workflows.

### LINKED ISSUE #482: [FEATURE] Load generator role
**Is your feature request related to a problem? Please describe.**
Currently, a load generator is deployed as part of some use cases, e.g. auto rem w/ Ansible. As this is a recurring use case, a dedicated role should be created that deploys a load generator for a configurable endpoint.

### LINKED ISSUE #488: [BUG] Several Gitlab installation and configuration issues
**Describe the bug**

## Gitlab depends on Keptn facts

Installation fails if Keptn isn't installed, i.e. facts `keptn_endpoint`, `keptn_bridge` and `keptn_api_token` not set and `cloud_automation_flavor == 'KEPTN'` (default).

## Gitlab _source-secret_ task misses `gitlab_internal_endpoint`

Required variables not defined when task is run standalone, e.g. without prior running _configure_.

## Gitlab _ensure-group_ task misses `gitlab_internal_endpoint` and `gitlab_oauth_token`

Required variables not defined when task is run standalone, e.g. without prior running _configure_.

## Gitlab _ensure-group_ task doesn't handle multiple pre-existing groups

While checking whether a group exists, task always assumes first object in response is required group. Logic breaks if there are multiple objects.

## Gitlab _ensure-project_ task sources `gitlab_internal_endpoint` and `gitlab_oauth_token` in reverse order

Currently, OAuth token is sourced prior to sourcing endpoint which fails, e.g. without prior running _configure_.

## Gitlab _ensure-project_ task doesn't handle multiple pre-existing projects

While checking whether a project exists, task always assumes first object in response is required project. Logic breaks if there are multiple objects.

### PR DESCRIPTION
**Pull request by tobigremmer-dt**
Resolves #477

### CODE REVIEW

**Comment by tobigremmer-dt** [comment]
Resolves #482

**Comment by drqc** [comment]
I spin up the use case, there is a part missing about authorizing the external URL in the platform, otherwise you will get this error
image

**Comment by drqc** [comment]
another error that I'm getting
image

**Comment by tobigremmer-dt** [comment]
> another error that I'm getting
> image

Resolved.

**Comment by tobigremmer-dt** [comment]
> I spin up the use case, there is a part missing about authorizing the external URL in the platform, otherwise you will get this error
> image

Resolved.

**Comment by tobigremmer-dt** [comment]
Resolves #488

**Comment by drqc** [comment]
i tested the use case. A couple of comments,
- Regarding the workflow documentation, i would add more details explaining how the workflow works since it's the main part of the remediation. (i.e. why the DQL expression works to filter the event, what is the get_owners (since the docs doesnt' mention that I have to add owners not sure how it works or how to configure ownership for the demo, (i see this automatically done but since it's new I would add some docs )
image
- I would also add the remediation status as a comment to the Dynatrace Problem card as it's the most visible way to realized why the problem has closed. (similar to the auto-remediation with ansible)
- Also docs about the workflow getting executed 2 times (for problem open and problem closed would help understand the use case).

**Comment by tobigremmer-dt** [comment]
- [x] Workflow details
- [x] DT Problem comment
- [x] Open/closed workflow executions

**Comment by tobigremmer-dt** [comment]
@drqc All done! Would appreciate a review.

File: user-skel/ansible/roles/demo-ar-workflows-gitlab/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+|dt_oauth_client_id|Dynatrace OAuth client id. Make sure scopes `automation:workflows:read` and `automation:workflows:write` are assigned to your OAuth client|
+|dt_oauth_client_secret|Dynatrace OAuth client secret|
+|dt_oauth_account_urn|Dynatrace OAuth account URN|
+
+Extra vars can be set e.g. as Terraform variables:
+
+```
+extra_vars = {
+  dt_environment_url_gen3 = "https://<YOUR ENVIRONMENT ID>.sprint.apps.dynatracelabs.com"
+  dt_oauth_sso_endpoint   = "https://sso-sprint.dynatracelabs.com"
```
Comment (drqc) [inline user-skel/ansible/roles/demo-ar-workflows-gitlab/README.md]: For the jenkins oauth vars I'm using https://sso-sprint.dynatracelabs.com/sso/oauth2/token with the sso/oauth2/token additional URL. I can update the vars to match this one for convention

**Review by drqc — APPROVED** [review]
lgtm

**Pull request merged**